src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -167,6 +168,7 @@ public CalledHaplotypes assignGenotypeLikelihoods(final List<Haplotype> haplotyp
                             loc);
 
             VariantContext mergedVC = AssemblyBasedCallerUtils.makeMergedVariantContext(eventsAtThisLocWithSpanDelsReplaced);
+            int mergedAlleles = mergedVC.getAlleles().size();
```

Comment:
You deference mergedVC here, but immediately afterwards there is a check for mergedVC == null.  I think you've introduced an NPE here.